Verilator build fails in memutils

```
../src/lowrisc_systems_top_earlgrey_verilator_0.1/top_earlgrey_verilator.cc:8:10: fatal error: verilator_memutil_extension.h: No such file or directory
 #include "verilator_memutil_extension.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=24817&view=logs&j=a8b61b66-4668-57fb-dcc3-ba755ef807ae&t=d8303847-c8da-507b-1651-4a69545a97af

Fallout from #3883 most likely.

@rswarbrick can you have a look?

--- a/hw/dv/verilator/cpp/dpi_memutil.cc
+++ b/hw/dv/verilator/cpp/dpi_memutil.cc
@@ -375,8 +375,12 @@ void StagedMem::AddSegment(uint32_t offset, std::vector<uint8_t> &&seg) {
 }
 
 std::vector<uint8_t> StagedMem::GetFlat() const {
-  size_t len = (size_t)max_addr_ - min_addr_;
+  // Since max_addr_ and min_addr_ are inclusive, the size to allocate
+  // is 1+(max-min). We cast to size_t to make sure the +1 doesn't
+  // overflow.
+  size_t len = (size_t)1 + (max_addr_ - min_addr_);
   std::vector<uint8_t> ret(len, 0);
+
   for (const auto &pr : segs_) {
     const AddrRange<uint32_t> &rng = pr.first;
     const std::vector<uint8_t> &seg = pr.second;

--- a/hw/top_earlgrey/top_earlgrey_verilator.cc
+++ b/hw/top_earlgrey/top_earlgrey_verilator.cc
@@ -5,12 +5,12 @@
 #include <iostream>
 
 #include "verilated_toplevel.h"
-#include "verilator_memutil_extension.h"
+#include "verilator_memutil.h"
 #include "verilator_sim_ctrl.h"
 
 int main(int argc, char **argv) {
   top_earlgrey_verilator top;
-  VerilatorMemUtilExtension memutil;
+  VerilatorMemUtil memutil;
   VerilatorSimCtrl &simctrl = VerilatorSimCtrl::GetInstance();
   simctrl.SetTop(&top, &top.clk_i, &top.rst_ni,
                  VerilatorSimCtrlFlags::ResetPolarityNegative);

